pkg/cmd/run.go

```diff
@@ -411,7 +411,15 @@ func fetchSecrets(localConfig models.ScopedOptions, enableCache bool, fallbackOp
 
 	statusCode, respHeaders, response, httpErr := http.DownloadSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, format, nameTransformer, etag, dynamicSecretsTTL, secretNames)
 	if !httpErr.IsNil() {
-		if fallbackOpts.enable {
+		canUseFallback := statusCode != 401 && statusCode != 403
```

Comment:
We should also delete the fallback if it's a 404 error in case the config or project was deleted.

Reply:
That sounds good to me; @Piccirello are you aware of a reason we shouldn't do that?

Reply:
@apazzolini, @nmanoogian and I were talking about this and one suggestion was to only use the fallback file for network errors and 5xx status codes. Or in other words, do not use the fallback file for 4xx errors. One reason this is problematic is that we send `400` for all errors, even sanity checks, unless we explicitly use another error code.

Reply:
There's also the option of just not using the fallback file (instead of deleting it) for 4xx errors. I don't know that I have a super strong opinion on the best thing to do here.

Reply:
On the surface, I can get behind the idea of not using the fallback file on any 4xx error. My only worry is that in the event of a bug on our end, we're most likely to erroneously send a 400. This would mean that the fallback file wouldn't be used, making the error significantly more severe. Thus I'm leaning towards having a specific allowlist of error codes that lead us to ignore the fallback file (and delete it). It would be anything related to authentication/authorization/you no longer having access to secrets. `401`, `403`, and `404` seem most appropriate to me.

Reply:
That SGTM. Updated to include `404`.